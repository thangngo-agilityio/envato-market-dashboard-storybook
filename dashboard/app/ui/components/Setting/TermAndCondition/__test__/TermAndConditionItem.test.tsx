import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

// Components
import TermAndConditionItem from '../TermAndConditionItem';

describe('TermAndConditionItem component', () => {
  it('should renders correctly', () => {
    const { container } = render(
      <TermAndConditionItem heading="heading" content="content" />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should renders heading and content correctly', () => {
    render(<TermAndConditionItem heading="heading" content="content" />);

    // Assert that the question is rendered correctly
    const questionElement = screen.getByText('heading');
    expect(questionElement).toMatchSnapshot('heading');

    // Assert that the answer is not initially visible
    const answerElement = screen.queryByText('content');
    expect(answerElement).toMatchSnapshot('content');
  });
});
