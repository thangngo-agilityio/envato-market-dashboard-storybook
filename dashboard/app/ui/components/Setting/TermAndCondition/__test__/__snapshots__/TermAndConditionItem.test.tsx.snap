// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TermAndConditionItem component should renders correctly 1`] = `
<div>
  <div
    class="css-19idom"
  >
    <h4
      class="chakra-text css-1gmhtzj"
    >
      heading
    </h4>
    <p
      class="chakra-text css-1wtnvd2"
    >
      content
    </p>
    <p
      class="chakra-text css-b0xv85"
    />
  </div>
</div>
`;

exports[`TermAndConditionItem component should renders heading and content correctly: content 1`] = `
<p
  class="chakra-text css-1wtnvd2"
>
  content
</p>
`;

exports[`TermAndConditionItem component should renders heading and content correctly: heading 1`] = `
<h4
  class="chakra-text css-1gmhtzj"
>
  heading
</h4>
`;
